src/screens/ComposeMessageScreen.test.tsx

```diff
@@ -0,0 +1,211 @@
+              tileId: 'mockTileId',
+            },
+          } as any
+        }
+      />
+    </GraphQLClientContextProvider>,
+  );
+  const searchBar = getByTestId('search-bar');
+  fireEvent.changeText(searchBar, 'User');
+  expect(queryAllByTestId('user-list-item').length).toBe(10);
+});
+
+test('single item returned when searched', () => {
+  const { queryAllByTestId, getByTestId } = render(
+    <GraphQLClientContextProvider baseURL={baseURL}>
+      <ComposeMessageScreen
+        navigation={useNavigation() as any}
+        route={
+          {
+            params: {
+              tileId: 'mockTileId',
+            },
+          } as any
+        }
+      />
+    </GraphQLClientContextProvider>,
+  );
+  const searchBar = getByTestId('search-bar');
+  fireEvent.changeText(searchBar, 'User 1');
+  expect(queryAllByTestId('user-list-item').length).toBe(1);
+});
+
+test('item added as chip but removed from search when selected', () => {
+  const { queryAllByTestId, getByTestId } = render(
+    <GraphQLClientContextProvider baseURL={baseURL}>
+      <ComposeMessageScreen
+        navigation={useNavigation() as any}
+        route={
+          {
+            params: {
+              tileId: 'mockTileId',
+            },
+          } as any
+        }
+      />
+    </GraphQLClientContextProvider>,
+  );
+  const searchBar = getByTestId('search-bar');
+  fireEvent.changeText(searchBar, 'User 1');
+
+  const item = getByTestId('user-list-item');
+  expect(queryAllByTestId('chip').length).toBe(0);
+  fireEvent.press(item);
+  expect(queryAllByTestId('user-list-item').length).toBe(0);
+  expect(queryAllByTestId('chip').length).toBe(1);
+});
+
+test('compose button disabled by default', () => {
+  const { getByTestId } = render(
+    <GraphQLClientContextProvider baseURL={baseURL}>
+      <ComposeMessageScreen
+        navigation={useNavigation() as any}
+        route={
+          {
+            params: {
+              tileId: 'mockTileId',
+            },
+          } as any
+        }
+      />
+    </GraphQLClientContextProvider>,
+  );
+  const sendButton = getByTestId('send-button');
+  fireEvent.press(sendButton);
+  expect(replaceMock).not.toHaveBeenCalled();
+});
+
+test('compose button disabled with user selected but no message text', () => {
+  const { getByTestId } = render(
+    <GraphQLClientContextProvider baseURL={baseURL}>
+      <ComposeMessageScreen
+        navigation={useNavigation() as any}
+        route={
+          {
+            params: {
+              tileId: 'mockTileId',
+            },
+          } as any
+        }
+      />
+    </GraphQLClientContextProvider>,
+  );
+  const textInput = getByTestId('Type a message...');
+  fireEvent.changeText(textInput, 'some new message');
+
+  const sendButton = getByTestId('send-button');
+  fireEvent.press(sendButton);
+  expect(replaceMock).not.toHaveBeenCalled();
+});
+
+test('compose button disabled with message text but no user selected', async () => {
```

Comment:
Test title doesn't seem to match the test

Reply:
Copy/paste, will fix